precompiles/parachain-staking/src/tests.rs

```diff
@@ -191,6 +191,161 @@ fn points_non_zero() {
 		});
 }
 
+#[test]
+fn delegation_amount_zero() {
+	ExtBuilder::default()
+		.with_balances(vec![
+			(TestAccount::Alice, 1_000),
+			(TestAccount::Charlie, 50),
+			(TestAccount::Bogus, 50),
+		])
+		.with_candidates(vec![(TestAccount::Alice, 1_000)])
+		.with_delegations(vec![
+			(TestAccount::Charlie, TestAccount::Alice, 50),
+			(TestAccount::Bogus, TestAccount::Alice, 50),
+		])
+		.build()
+		.execute_with(|| {
+			let selector = &Keccak256::digest(b"delegation_amount(address,address)")[0..4];
+
+			// Construct data to read delegation amount
+			let mut input_data = Vec::<u8>::from([0u8; 68]);
+			input_data[0..4].copy_from_slice(&selector);
+			input_data[16..36].copy_from_slice(&TestAccount::Alice.to_h160().0);
+			input_data[48..68].copy_from_slice(&TestAccount::Bob.to_h160().0);
+
+			// Expected result 3
+			let expected_one_result = Some(Ok(PrecompileOutput {
+				exit_status: ExitSucceed::Returned,
+				output: EvmDataWriter::new().write(U256::zero()).build(),
+				cost: Default::default(),
+				logs: Default::default(),
+			}));
+
+			// Assert that there is 0 delegation amount
+			assert_eq!(
+				Precompiles::execute(precompile_address(), &input_data, None, &evm_test_context()),
+				expected_one_result
+			);
+		});
+}
+
+#[test]
+fn delegation_amount_nonzero() {
+	ExtBuilder::default()
+		.with_balances(vec![
+			(TestAccount::Alice, 1_000),
+			(TestAccount::Charlie, 50),
+			(TestAccount::Bogus, 50),
+		])
+		.with_candidates(vec![(TestAccount::Alice, 1_000)])
+		.with_delegations(vec![
+			(TestAccount::Charlie, TestAccount::Alice, 50),
+			(TestAccount::Bogus, TestAccount::Alice, 50),
+		])
+		.build()
+		.execute_with(|| {
+			let selector = &Keccak256::digest(b"delegation_amount(address,address)")[0..4];
+
+			// Construct data to read delegation amount
+			let mut input_data = Vec::<u8>::from([0u8; 68]);
+			input_data[0..4].copy_from_slice(&selector);
+			input_data[16..36].copy_from_slice(&TestAccount::Alice.to_h160().0);
+			input_data[48..68].copy_from_slice(&TestAccount::Charlie.to_h160().0);
+			// Amount of Charlie's delegation in support of Alice
+			let expected_amt: U256 = 50u32.into();
+			let expected_result = Some(Ok(PrecompileOutput {
+				exit_status: ExitSucceed::Returned,
+				output: EvmDataWriter::new().write(expected_amt).build(),
+				cost: Default::default(),
+				logs: Default::default(),
+			}));
+
+			// Assert that there is delegation amount of 50
+			assert_eq!(
+				Precompiles::execute(precompile_address(), &input_data, None, &evm_test_context()),
+				expected_result
+			);
+		});
+}
+
+#[test]
+fn is_in_top_delegations_false() {
```

Comment:
There are two different code paths that can make this return false. One is where there is a delegation, but it is too small to be in the top. The other is when there is no delegation at all. It would be good to test both paths.